controllers/idler/idler_controller_test.go

```diff
@@ -448,6 +547,188 @@ func TestEnsureIdlingFailed(t *testing.T) {
+		err := reconciler.createNotification(logf.FromContext(context.TODO()), idler)
+		//then
+		require.NoError(t, err)
+		require.True(t, condition.IsTrue(idler.Status.Conditions, toolchainv1alpha1.IdlerTriggeredNotificationCreated))
+		//check notification was created
+		hostCl, _ := reconciler.GetHostCluster()
+		notification := toolchainv1alpha1.Notification{}
+		err = hostCl.Client.Get(context.TODO(), types.NamespacedName{Name: "alex-stage-idled", Namespace: hostCl.OperatorNamespace}, &notification)
+		require.NoError(t, err)
+		createdTime := notification.CreationTimestamp
+
+		t.Run("Notification not created if already sent", func(t *testing.T) {
+			//when
+			err = reconciler.createNotification(logf.FromContext(context.TODO()), idler)
+			//then
+			require.NoError(t, err)
+			err = hostCl.Client.Get(context.TODO(), types.NamespacedName{Name: "alex-stage-idled", Namespace: hostCl.OperatorNamespace}, &notification)
+			require.NoError(t, err)
+			require.Equal(t, createdTime, notification.CreationTimestamp)
+		})
+	})
+
+	t.Run("Creates notification when notification creation had failed previously", func(t *testing.T) {
+		idler.Status.Conditions = []toolchainv1alpha1.Condition{
+			{
+				Type:    toolchainv1alpha1.IdlerTriggeredNotificationCreated,
+				Status:  corev1.ConditionFalse,
+				Reason:  toolchainv1alpha1.IdlerTriggeredNotificationCreationFailedReason,
+				Message: "notification wasn't created before",
+			},
+		}
+		namespaces := []string{"dev", "stage"}
+		usernames := []string{"alex"}
+		nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "alex", "advanced", "abcde11", namespaces, usernames)
+		mur := newMUR("alex")
+		reconciler, _, _, _ := prepareReconcile(t, idler.Name, newGetHostClusterReady, idler, nsTmplSet, mur)
+
+		//when
+		err := reconciler.createNotification(logf.FromContext(context.TODO()), idler)
+		//then
+		require.NoError(t, err)
+		require.True(t, condition.IsTrue(idler.Status.Conditions, toolchainv1alpha1.IdlerTriggeredNotificationCreated))
+	})
+
+	t.Run("Condition is set when setting failed previously, and notification is only sent once", func(t *testing.T) {
+		//This is to check the scenario when setting condition fails after creating a notification. Second reconcile expects to attempt the condition again, but not create the notification
+		// given
+		idler.Status.Conditions = nil
+		namespaces := []string{"dev", "stage"}
+		usernames := []string{"alex"}
+		nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "alex", "advanced", "abcde11", namespaces, usernames)
+		mur := newMUR("alex")
+		reconciler, _, cl, _ := prepareReconcile(t, idler.Name, newGetHostClusterReady, idler, nsTmplSet, mur)
+		cl.MockStatusUpdate = func(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+			return errors.New("can't update condition")
+		}
+		//when
+		err := reconciler.createNotification(logf.FromContext(context.TODO()), idler)
+
+		//then
+		require.Error(t, err, "can't update condition")
```

Comment:
```suggestion
		require.EqualError(t, err, "can't update condition")
```

Reply:
done in [67a0dde](https://github.com/codeready-toolchain/member-operator/pull/357/commits/67a0dde897ead65c14667363a42ff694ae962c68)